source/dofs/dof_tools.cc

```diff
@@ -1144,6 +1144,29 @@ namespace DoFTools
     Assert(dof_handler.n_dofs() > 0,
            ExcMessage("Number of DoF is not positive. "
                       "This could happen when the function is called before NumberCache is written."));
+
+    // in case this function is executed with parallel::shared::Triangulation
+    // and artifical cells, we need to store owners of cells separately
+    std::vector<unsigned int> cell_owners (dof_handler.get_tria().n_active_cells());
+    for (typename DH::active_cell_iterator cell = dof_handler.begin_active();
+         cell!= dof_handler.end(); cell++)
+      if (cell->is_locally_owned())
+        cell_owners[cell->active_cell_index()] = cell->subdomain_id();
```

Comment:
Can you add a comment that states that the vector at the end of this loop only contains nonzero entries for all locally owned cells, and zero for all other cells?